pay-api/src/pay_api/services/base_payment_system.py

```diff
@@ -259,18 +298,23 @@ def _refund_and_create_credit_memo(invoice: InvoiceModel):
         CreditModel(
             cfs_identifier=cms_response.get("credit_memo_number"),
             is_credit_memo=True,
-            amount=invoice.total,
-            remaining_amount=invoice.total,
+            amount=refund_amount,
+            remaining_amount=refund_amount,
             account_id=invoice.payment_account_id,
         ).flush()
 
         # Add up the credit amount and update payment account table.
         payment_account: PaymentAccountModel = PaymentAccountModel.find_by_id(invoice.payment_account_id)
-        payment_account.credit = (payment_account.credit or 0) + invoice.total
+        payment_account.credit = (payment_account.credit or 0) + refund_amount
         current_app.logger.info(
-            f"Updating credit amount to  {payment_account.credit} for account {payment_account.auth_account_id}"
+            f"Updating credit amount to {payment_account.credit} for account {payment_account.auth_account_id}"
         )
         payment_account.flush()
+
+        # For partial refunds that aren't for the full amount, keep as PAID
```

Comment:
Obvious if you read the next 2 lines

You're just making yourself more work here. Imagine refactoring with comments like these? you'd have to change the code and the comments.